pkg/operator/encryption/deployer/staticpod.go

```diff
@@ -0,0 +1,173 @@
+package deployer
+
+import (
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/informers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/encryption/encryptionconfig"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/encryption/statemachine"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+type StaticPodDeployer struct {
+	podClient                corev1client.PodInterface
+	secretClient             corev1client.SecretInterface
+	targetNamespaceInformers informers.SharedInformerFactory
+}
+
+var _ statemachine.Deployer = &StaticPodDeployer{}
+
+// NewStaticPodDeployer create a deployer abstraction meant for the staticpod controllers. It copies
+// the encryption-config-<targetNamespace> from openshift-config-managed namespace to the target namespace
+// as encryption-config. From there it is revisioned and deployed to the static pods. The last
+// deployed encryption config is read from encryption-config-<revision>.
+func NewStaticPodDeployer(
+	targetNamespace string,
+	informers operatorv1helpers.KubeInformersForNamespaces,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+	podClient corev1client.PodsGetter,
+	secretClient corev1client.SecretsGetter,
+) (*StaticPodDeployer, error) {
+	if resourceSyncer != nil {
+		if err := resourceSyncer.SyncSecret(
+			resourcesynccontroller.ResourceLocation{Namespace: targetNamespace, Name: encryptionconfig.EncryptionConfSecretName},
+			resourcesynccontroller.ResourceLocation{Namespace: operatorclient.GlobalMachineSpecifiedConfigNamespace, Name: fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, targetNamespace)},
+		); err != nil {
+			return nil, err
+		}
+	}
+
+	return &StaticPodDeployer{
+		podClient:                podClient.Pods(targetNamespace),
+		secretClient:             secretClient.Secrets(targetNamespace),
+		targetNamespaceInformers: informers.InformersFor(targetNamespace),
+	}, nil
+}
+
+// DeployedEncrytionConfigSecret returns the deployed encryption config and whether all
+// instances of the operand have acknowledged it.
+func (d *StaticPodDeployer) DeployedEncrytionConfigSecret() (secret *corev1.Secret, converged bool, err error) {
```

Comment:
typo: DeployedEncryptionConfigSecret

Reply:
my fingers don't like the word encryption.

Reply:
fixed